pkg/reconciler/managed/reconciler.go

```diff
@@ -367,11 +367,8 @@ func defaultMRManaged(m manager.Manager) mrManaged {
 	return mrManaged{
 		ConnectionPublisher: NewAPISecretPublisher(m.GetClient(), m.GetScheme()),
 		Finalizer:           resource.NewAPIFinalizer(m.GetClient(), managedFinalizerName),
-		Initializer: InitializerChain{
-			NewDefaultProviderConfig(m.GetClient()),
```

Comment:
I considered leaving this out of an abundance of caution, because:

* It's a no-op when the ProviderConfigReference is already set.
* Many providers assume the ProviderConfigReference will never be nil
* It's possible that a provider will be built against this library, but forget to generate/apply new CRDs.

I decided to just remove it, given that it's very unlikely anyone using the package manager and our standard build tooling would ever end up accidentally deploying new provider code without also updating their CRDs.